templates/cluster-template-without-lb.yaml

```diff
@@ -0,0 +1,199 @@
+---
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: Cluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: ["192.168.0.0/16"] # CIDR block used by Calico.
+    serviceDomain: "cluster.local"
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: OpenStackCluster
+    name: ${CLUSTER_NAME}
+  controlPlaneRef:
+    kind: KubeadmControlPlane
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+    name: ${CLUSTER_NAME}-control-plane
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: OpenStackCluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  cloudName: ${OPENSTACK_CLOUD}
+  cloudsSecret:
+    name: ${CLUSTER_NAME}-cloud-config
+    namespace: ${CLUSTER_NAME}
+  controlPlaneEndpoint:
+    host: ${OPENSTACK_CONTROLPLANE_IP}
+    port: 6443
+  nodeCidr: 10.6.0.0/24
+  dnsNameservers:
+  - ${OPENSTACK_DNS_NAMESERVERS}
+  externalNetworkId: ${OPENSTACK_EXTERNAL_NETWORK_ID}
+  disablePortSecurity: true
+  disableServerTags: true
+  useOctavia: false
+---
+kind: KubeadmControlPlane
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+metadata:
+  name: "${CLUSTER_NAME}-control-plane"
+spec:
+  replicas: 1
+  infrastructureTemplate:
+    kind: OpenStackMachineTemplate
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    name: "${CLUSTER_NAME}-control-plane"
+  kubeadmConfigSpec:
+    initConfiguration:
+      nodeRegistration:
+        name: '{{ ds.meta_data.name }}'
+        kubeletExtraArgs:
+          cloud-provider: openstack
+          cloud-config: /etc/kubernetes/cloud.conf
+    clusterConfiguration:
+      controlPlaneEndpoint: "${OPENSTACK_CONTROLPLANE_IP}:6443"
+      imageRepository: k8s.gcr.io
+      apiServer:
+        extraArgs:
+          cloud-provider: openstack
+          cloud-config: /etc/kubernetes/cloud.conf
+        extraVolumes:
+        - name: cloud
+          hostPath: /etc/kubernetes/cloud.conf
+          mountPath: /etc/kubernetes/cloud.conf
+          readOnly: true
+      controllerManager:
+        extraArgs:
+          cloud-provider: openstack
+          cloud-config: /etc/kubernetes/cloud.conf
+        extraVolumes:
+        - name: cloud
+          hostPath: /etc/kubernetes/cloud.conf
+          mountPath: /etc/kubernetes/cloud.conf
+          readOnly: true
+        - name: cacerts
+          hostPath: /etc/certs/cacert
+          mountPath: /etc/certs/cacert
+          readOnly: true
+    joinConfiguration:
+      nodeRegistration:
+        name: '{{ ds.meta_data.name }}'
```

Comment:
Could you teach me why local_hostname changed to ds.meta_data.name?

Reply:
I tried to use `ds.meta_data.local_hostname` like capa. But that one wasn't set in my devstack. So I though `ds.metadata` might be better then just `local_hostname`. But I can switch it back to `local_hostname` if you want. I only want something which works in all of our cases. Not sure what the best one is.

Reply:
Yes, I also tried using ```ds.meta_data.local_hostname``` but it includes domain name like novalocal which is not compatible with OpenStack cloud provider.  OpenStack cloud provider tried to find host without domain name and failed. Instead, ```local_hostname``` does not include domain name so was successfully ended. ```ds.meta_data.name``` is also missing domain name so that is would work.

Reply:
Fixed (sorry I quashed for the final merge, but just check the templates :))